Title: Cloud-init bi-weekly status
Date: 2021-03-09 17:19:00 UTC
Category: status-meeting-minutes
Slug: status-2021-03-09

Meeting information
----------------

 * \#cloud-init: cloud-init bi-weekly office-hours meeting, started by blackboxsw, 09 Mar at 17:19 &mdash; 18:38 UTC.
 * Full logs at [https://new.ubottu.com/meetingology/logs/cloud-init/2021/cloud-init.2021-03-09-17.19.log.html](https://new.ubottu.com/meetingology/logs/cloud-init/2021/cloud-init.2021-03-09-17.19.log.html)



Meeting summary
----------------

 *  **LINK**: [https://discourse.ubuntu.com/t/cloud-init-status-03-09-2021/21309](https://discourse.ubuntu.com/t/cloud-init-status-03-09-2021/21309)   (blackboxsw, 17:23)



People present (lines said)
----------------

 * blackboxsw (14)
 * falcojr (8)
 * beantaxi (7)
 * meetingology (3)
 * Odd_Bloke (3)



Full log
----------------


 17:19 `<blackboxsw>` \#startmeeting cloud-init bi-weekly office-hours

 17:19 `<meetingology>` Meeting started at 17:19:08 UTC.  The chair is blackboxsw.  Information about MeetBot at [https://wiki.ubuntu.com/meetingology](https://wiki.ubuntu.com/meetingology)

 17:19 `<meetingology>` Available commands: action, commands, idea, info, link, nick

 17:19 `<blackboxsw>` \#char Odd_Bloke falcojr

 17:19 `<blackboxsw>` \#chair Odd_Bloke falcojr

 17:19 `<meetingology>` Current chairs: Odd_Bloke, blackboxsw, falcojr

 17:20 `<blackboxsw>` hi cloud-init, sorry for the confusion on the UTC time that I posted in the topic. I had incorrectly set 15:15 UTC instead of 17:15.

 17:20 `<blackboxsw>` welcome to the bi-weekly cloud-init office hours

 17:21 `<blackboxsw>` This cloud-init status meeting has moved to a more asynchronous posting of high-level happenings in cloud-init to discourse to support timezone differences for all cloud-init developers.

 17:23 `<blackboxsw>` We plan on continuing to host these office hours to provide a means for quick communication, questions and discussions with a couple of upstream developers present to help facilite any questions or discussions of interest to folks.

 17:23 `<blackboxsw>` For reference, here are the high-level "happenings" in cloud-init upstream over the last two weeks.

 17:23 `<Odd_Bloke>` o/

 17:23 `<blackboxsw>` \#link [https://discourse.ubuntu.com/t/cloud-init-status-03-09-2021/21309](https://discourse.ubuntu.com/t/cloud-init-status-03-09-2021/21309)

 17:26 `<blackboxsw>` As of last office-hours I belive upstream had just cut the first 21.1 release in the tip of cloud-init. Canonical will be starting the SRU process this week I believe to publish the 21.1 release (plus a couple of extra fixes which just landed in tip) into Ubuntu Xenial, Bionic, Focal, Groovy and Hirsute.

 17:41 `<beantaxi>` Odd_Bloke: I tried your command and it was helpful. It complained about a missing ssh key, so I shrugged, tried ssh-keygen -t rsa and reran the test, and it passed. I've kicked off a run of all the tests, which I assume will take a bit.

 17:55 `<Odd_Bloke>` beantaxi: Great, thanks for the update!  A full run in LXD containers takes ~12mins locally (though I'm probably running on beefier hardware than you're paying EC2 for so YMMV :p).

 17:55 `<Odd_Bloke>` beantaxi: Once you had the output, was it clear that the missing SSH key was the issue?

 17:55 `<blackboxsw>` setting the appropriate time for next office hours

 17:56 `<blackboxsw>` setting the appropriate time for next office hours Mar 23 17:15 UTC

 17:57 `<beantaxi>` ERROR tests/integration_tests/modules/test_runcmd.py::TestRuncmd::test_runcmd - FileNotFoundError: [Errno 2] No such file or directory: '/home/ubuntu/.ssh/id_rsa.pub'

 17:57 `<beantaxi>` That was the error message

 17:59 `<falcojr>` tests require an ssh keypair. It'll use your user one by default or you can specify one manually in the integration settings config

 17:59 `<falcojr>` do an `ssh-keygen`

 17:59 `<beantaxi>` The full test failed with 7 failures: [https://paste.ubuntu.com/p/hcYN7ZDvbs/](https://paste.ubuntu.com/p/hcYN7ZDvbs/)

 18:00 `<beantaxi>` falcojr: Thanks ... I was just showing Odd_Bloke the error message I got before running ssh-keygen, which did indeed solve most of my issues

 18:01 `<beantaxi>` falcojr: Also, this is just me running the existing integration tests on an EC2 instance. Without any of my changes

 18:01 `<falcojr>` not sure the cause of that first one, but the rest are for code not in ubuntu images yet

 18:03 `<falcojr>` the version of cloud_init to use is specified by the CLOUD_INIT_SOURCE variable (integration_settings.py...copy this to user_settings.py to easily override defaults)

 18:03 `<falcojr>` it defaults to NONE which means use whatever image has been released on this platform

 18:04 `<falcojr>` to test non-released code, changing it to IN_PLACE (only works on LXD) should work

 18:08 `<falcojr>` you can also set it to ppa:cloud-init-dev/daily , but that won't contain any changes that you're trying to test

 18:18 `<beantaxi>` falcojr: Copying integration_settings.py to user_settings.py and setting CLOUD_INIT_SOURCE=IN_PLACE made all the test_keys_to_console errors clean. There's still that simple_streams error. Do I just need to add a PPA for simplestreams? It looks like requires a file in /etc/apt/sources.list.d, but mine is empty

 18:23 `<falcojr>` Not sure off the top of my head. Did you pip install the integration_requirements.txt?

 18:38 * blackboxsw thinks I should about wrap up office hours. Thanks falcojr Odd_Bloke and beantaxi for the discussion there

 18:38 `<blackboxsw>` \#endmeeting



Generated by MeetBot 0.4.0 ([https://wiki.ubuntu.com/meetingology](https://wiki.ubuntu.com/meetingology))